pkg/processor/k8sprocessor/kube/client.go

```diff
@@ -321,6 +322,9 @@ func (c *WatchClient) extractPodAttributes(pod *api_v1.Pod) map[string]string {
 	}
 
 	if c.Rules.NodeName {
+		if len(pod.Spec.NodeName) == 0 {
+			c.logger.Warn("missing Node name for Pod, cache may be out of sync", zap.String("pod", pod.Name))
```

Comment:
We need to watch if these logs are not too verbose.
Metrics for these would be a good solution too.

Reply:
I thought about metrics, but realistically this is a fault that won't happen very often. For this to occur, a Pod needs to get scheduled, start, emit logs, and the logs get collected and pass through the processor, and in the meantime the processor can't get any updates from the API Server. There's already similar warnings in client-go itself, they're just not very specific.